subsys/bluetooth/host/id.c

```diff
@@ -0,0 +1,1740 @@
+/*
+ * Copyright (c) 2017 Nordic Semiconductor ASA
+ * Copyright (c) 2015-2016 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <settings/settings.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci_vs.h>
+#include <bluetooth/buf.h>
+
+#include "hci_core.h"
+#include "id.h"
+#include "smp.h"
+#include "conn_internal.h"
+#include "keys.h"
+#include "common/rpa.h"
+
+#include "settings.h"
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_DEBUG_HCI_CORE)
+#define LOG_MODULE_NAME bt_id
+#include "common/log.h"
+
+#if defined(CONFIG_BT_OBSERVER) || defined(CONFIG_BT_BROADCASTER)
+const bt_addr_le_t *bt_lookup_id_addr(uint8_t id, const bt_addr_le_t *addr)
+{
+	if (IS_ENABLED(CONFIG_BT_SMP)) {
+		struct bt_keys *keys;
+
+		keys = bt_keys_find_irk(id, addr);
+		if (keys) {
+			BT_DBG("Identity %s matched RPA %s",
+			       bt_addr_le_str(&keys->addr),
+			       bt_addr_le_str(addr));
+			return &keys->addr;
+		}
+	}
+
+	return addr;
+}
+#endif /* CONFIG_BT_OBSERVER || CONFIG_BT_CONN */
+
+static void adv_id_check_func(struct bt_le_ext_adv *adv, void *data)
+{
+	struct bt_adv_id_check_data *check_data = data;
+
+	if (IS_ENABLED(CONFIG_BT_EXT_ADV)) {
+		/* Only check if the ID is in use, as the advertiser can be
+		 * started and stopped without reconfiguring parameters.
+		 */
+		if (check_data->id == adv->id) {
+			check_data->adv_enabled = true;
+		}
+	} else {
+		if (check_data->id == adv->id &&
+		    atomic_test_bit(adv->flags, BT_ADV_ENABLED)) {
+			check_data->adv_enabled = true;
+		}
+	}
+}
+
+#if defined(CONFIG_BT_PRIVACY)
+static void adv_is_private_enabled(struct bt_le_ext_adv *adv, void *data)
```

Comment:
It's just a complicated way to return, since the function has to follow the adv_foreach declaration, then the state contains the value returned.

This code is just moved.